### ISSUE TITLE
Problem building SparkCLR code with msbuild VS2015 settings on VSO

### ISSUE THREAD

**Issue by jthelin** [Microsoft/SparkCLR#8]
Any clues to why this code won't build with msbuild 2015 on VSO?

I set up an automated build from github master branch on my VSO instance with default settings for the msbuild step, and strangely the build failed with the errors below.

I knew the code built ok for me on my local machine (which has VS2013 installed), so as an experiment I set the VSO build step to use "VS2013" setting instead, and strangely the VSO build succeeded!

I then switched the VSO build step back to the "VS2015" setting and got the original failure again :(

I haven't got VS2015 on any of my machines yet, but I would guess that interactive build might show the same problem?

Any thoughts?

```
2015-10-31T17:54:51.2733808Z ##[error]csharp\Adapter\Microsoft.Spark.CSharp\Core\RDD.cs(258,109): Error CS0029: Cannot implicitly convert type 'T' to 'System.Collections.Generic.KeyValuePair<T, int>'
2015-10-31T17:54:51.2743806Z      2>Core\RDD.cs(258,109): error CS0029: Cannot implicitly convert type 'T' to 'System.Collections.Generic.KeyValuePair<T, int>' [C:\a\1\s\csharp\Adapter\Microsoft.Spark.CSharp\Adapter.csproj]
2015-10-31T17:54:51.2743806Z ##[error]csharp\Adapter\Microsoft.Spark.CSharp\Core\RDD.cs(258,109): Error CS1662: Cannot convert lambda expression to intended delegate type because some of the return types in the block are not implicitly convertible to the delegate return type
2015-10-31T17:54:51.2753809Z      2>Core\RDD.cs(258,109): error CS1662: Cannot convert lambda expression to intended delegate type because some of the return types in the block are not implicitly convertible to the delegate return type [C:\a\1\s\csharp\Adapter\Microsoft.Spark.CSharp\Adapter.csproj]
```

**Comment by xiongrenyi** [comment]
looks like explicit type conversion is required here by VS2005 compiler:


Map(x => new KeyValuePair<T, int>(x, 0)).ReduceByKey((x,y)
=> x, numPartitions).Map<T>(x => x.Key)



Date: Sat, 31 Oct 2015 11:07:50 -0700

**Comment by jthelin** [comment]
Fixed in PR #20
